src/pages/Home/components/Filter/LocationFilter/LocationFilter.tsx

```diff
@@ -0,0 +1,180 @@
+  const currentOs = getOS();
+
+  const devicesLinks: { [key: string]: string } = {
+    ['Android']:
+      'https://support.google.com/accounts/answer/3467281?hl=pt-BR#turn_on_off',
+    ['iOS']: 'https://support.apple.com/pt-br/102647',
+  };
+  const currentLink =
+    devicesLinks[currentOs] ||
+    'https://support.google.com/chrome/answer/142065?hl=pt-BR&co=GENIE.Platform%3DDesktop&sjid=11531706076940529676-SA';
+
+  return (
+    <Link to={currentLink} target="_blank" className="text-blue-600">
+      Veja mais informações sobre a opção de geolocalização do Iphone
+    </Link>
+  );
+};
+
+const LocationFilter = ({
+  geolocationFormValues,
+  geolocationValues,
+  setFieldValue,
+  error,
+}: ILocationFilter) => {
+  const {
+    isLoading,
+    geolocation,
+    getLocation,
+    error: errorGeolocation,
+  } = useGeolocation();
+  const [isProximityChecked, setIsProximityChecked] = useState<boolean>(
+    (Boolean(geolocationValues?.latitude) &&
+      Boolean(geolocationValues?.longitude)) ||
+      (Boolean(geolocationFormValues?.latitude) &&
+        Boolean(geolocationFormValues?.longitude))
+  );
+  const normalizedRadiusMeters =
+    geolocationFormValues?.radiusInMeters ??
+    geolocationValues?.radiusInMeters ??
+    0;
+
+  useEffect(() => {
+    if (!isProximityChecked) return;
+    if (!geolocation.latitude) return;
+    if (!geolocation.latitude) return;
+
+    setFieldValue(
+      'geolocation.radiusInMeters',
+      geolocationValues?.radiusInMeters ?? 0
+    );
+    setFieldValue('geolocation.latitude', geolocation.latitude);
+    setFieldValue('geolocation.longitude', geolocation.longitude);
+  }, [
+    geolocation.latitude,
+    geolocation.longitude,
+    geolocationValues?.radiusInMeters,
+    isProximityChecked,
+    setFieldValue,
+  ]);
+
+  const showFilter = isProximityChecked && !errorGeolocation && !isLoading;
+
+  return (
+    <div className="flex flex-col gap-2 w-full my-4">
+      <div>
+        <div className="flex flex-col gap-2 w-full">
+          <label className="flex items-center mb-2">
+            <input
+              type="checkbox"
+              className="mr-2 w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600"
+              onChange={(event) => {
+                const { checked } = event.target;
+                setIsProximityChecked(checked);
+                if (checked) {
+                  getLocation();
+                  return;
+                }
+
+                setFieldValue('geolocation.radiusInMeters', undefined);
+                setFieldValue('geolocation.latitude', undefined);
+                setFieldValue('geolocation.longitude', undefined);
+              }}
+              defaultChecked={isProximityChecked}
+            />
+            Filtro por proximidade
+          </label>
+
+          {Boolean(isLoading) && (
```

Comment:
isLoading já é boolean, não precisa de fazer o wrap

Reply:
Haha, força do hábito, obrigado!